### DISCUSSION TITLE
Sample Slots, Questions and Suggestions

### DISCUSSION THREAD

**Discussion by Syntonica** [bangcorrupt/hacktribe#205] [Q&A] [answered]
I haven't tried this. 

Yet.

However, a few questions/points first:
1. The moratorium on using the empty  slots before #501, is this due to the Hacktribe firmware itself? Or, are they being reserved for future expansion? There's a number of PCM I wish to keep from the original set and still have room for more user samples. (The slots not available in the original E2S firmware notwithstanding.)  I want to prep my .all file for when I change the firmware.
2. The biggest turnoff with the E2 is the menus. Can these be reconfigured? If so, setting the Shift-Pad shortcuts to our own settings would be nice indeed.  
3. Accessing a second IFX parameter with the Shift button would be awesome. Some of the muliple IFX could be collapsed into a single one, making the menu smaller.
4. Korg was one step away from having a song mode. Pattern Sets allow 64 pattern sets to be selected. A setting to allow these to play back sequentially would add a simple song mode.
5. Oh! and how many times can you reflash these guys before failure? I'm assuming it's thousands and I'm in no danger this century.

Anywat, I can't wait to try all the filters on my E2S!  Thanks for everybody's hard work.

**Comment by bangcorrupt** [comment, marked as answer]
1. It's buggy, so blocked for now as a workaround.  Hacktribe will load the *.all file from a different location than the factory firmware, so no harm having different sets of samples.  
2. #56.  There will updates to the UI, including shift functions.  The only barrier is printing to the screen.  
3.  [FX editing](https://github.com/bangcorrupt/hacktribe/wiki/MIDI#fx-edit) is coming.
4. Pattern Chains can do this.  I chain patterns in looping groups of 4, using Pattern Sets to jump to the first pattern of each group.  I'd like to expand the Event Recorder at some point.
5. [Flash datasheet](https://datasheet.octopart.com/MX25L12835FZ2I-10G-Macronix-datasheet-14372549.pdf) says 100,000 erase/program cycles and 20 years data retention.

**Comment by Syntonica** [comment]
Thanks for the quick reply.

1. The OG firmware randomly skips certain slots and will not load samples into them for some unfathomable reason.  Maybe to ward off end users from replacing the stock PCM? The whole system is  kind of weird how anything marked as "Analog" is not really a PCM file like the others, but some other hidden type. Are they multisamples, one per octave, to eliminare aliasing?

2 & 3. I honestly did try to read a number of discussions before posting. This is great news to hear that it's in the works.

4.  It works that way, of course, but I want a menu item, Play Song, where you set the beginning and end pads and Play will play the designated chains back in order.  I'm not a performer and I just want a simple way to export a whole song at once, without resorting to the Event recorder and trying to pay attention for however long. (I'm not exactly neurotypical.)

5. I've seen some chips rated a quite a few fewer flash cycles, but 100k flashes is more than enough for me, but I never heard of a life span before! Yikes! I better get cracking....

Anyway, it looks like you're already half way there fixing all the things I dislike about the E2.  But for a hands-on experience, it's just right amount of knobiness for me to get lost in...  Again, thanks so much for all of your hard work.
